chat.js

```diff
@@ -920,25 +925,54 @@ Chat.uncacheTree = function (root) {
 	} while (uncache.length > 0);
 };
 
-Chat.loadCommands = function () {
-	if (Chat.commands) return;
+Chat.loadPlugins = function () {
+	if (Chat.commands || Chat.filter) return;
 
 	FS('package.json').readTextIfExists().then(data => {
 		if (data) Chat.package = JSON.parse(data);
 	});
 
 	let baseCommands = Chat.baseCommands = require('./chat-commands').commands;
 	let commands = Chat.commands = Object.assign({}, baseCommands);
+	let chatfilters = Chat.filters = [];
 
-	// Install plug-in commands
+	// Install plug-in commands and chat filter
 
 	// info always goes first so other plugins can shadow it
 	Object.assign(commands, require('./chat-plugins/info').commands);
 
 	for (let file of FS('chat-plugins/').readdirSync()) {
 		if (file.substr(-3) !== '.js' || file === 'info.js') continue;
-		Object.assign(commands, require('./chat-plugins/' + file).commands);
+		const plugin = require(`./chat-plugins/${file}`);
+
+		Object.assign(commands, plugin.commands);
+
+		const filter = plugin.chatfilter;
+		if (filter) {
+			if (typeof filter !== 'function') {
+				require('./crashlogger')(`This chatfilter is not a function`, `Loading a chatfilter`, {
+					file: `File location: ../chat-plugins/${file}`,
+					filter: filter,
+				});
+			} else { // only use chatfilters that are a function
+				chatfilters.push(filter);
+			}
+		}
 	}
+
+	Chat.filter = function (message, user, room, connection, targetUser) {
+		if (!chatfilters.length) return message;
```

Comment:
I changed the check where it calls for ``Chat.filter`` to just check the length of ``Chat.filters`` instead, so I removed this line here now.